Source/WTF/wtf/RunLoop.cpp

```diff
@@ -169,15 +170,17 @@ void RunLoop::dispatch(Function<void()>&& function)
         wakeUp();
 }
 
-void RunLoop::dispatchAfter(Seconds delay, Function<void()>&& function)
+Ref<RunLoop::DispatchTimer> RunLoop::dispatchAfter(Seconds delay, Function<void()>&& function)
 {
     RELEASE_ASSERT(function);
-    auto timer = new DispatchTimer(*this);
-    timer->setFunction([timer, function = WTFMove(function)] {
+    Ref<DispatchTimer> timer = adoptRef(*new DispatchTimer(*this));
+    timer->setFunction([timer = timer.copyRef(), function = WTFMove(function)] {
+        Ref<DispatchTimer> protectedTimer { timer.copyRef() };
```

Comment:
Move timer to this. Not copying. Otherwise, this function is still keeping timer ref even after execution of this lambda.
This WTFMove and protectedTimer is to delete this lambda -> timer reference after running.
```
Ref<DispatchTimer> protectedTimer { WTFMove(timer); };
```
The reason of this is avoiding creating cyclic reference as much as possible to avoid memory leak.